lib/player_ai_minimax.rb

```diff
@@ -0,0 +1,60 @@
+class PlayerAiMinimax
+  attr_reader :mark
+
+  def initialize(mark, rules)
+    @mark = mark
+    @rules = rules
+    @corners = [0,2,6,8]
```

Comment:
Setting aside the fact that this is hardcoded, this could be moved to either a constant or its own method. For example

```
class PlayerAIMinimax
  CORNERS = [0,2,6,8]           # as a constant

  def initialize(mark, rules)
    @mark = mark
    @rules = rules
  end

  # ...

  private

  def corners                   # as a method
    @corners ||= [0,2,6,8]
  end
end
```

Main advantage IMO is it keeps the `initialize` method focused on what needs to be passed in to create the instance.